### ISSUE TITLE
Add an Android port

### ISSUE THREAD

**Issue by dezmen3** [ZenoArrows/The-Simpsons-Hit-and-Run#31]
Tried on Yuzu, EggNs and Skyline, it shows splash screen and then just crashes to emulator menu.
Is it working on android emulator or some special installation steps needs to be made?

**Comment by ZenoArrows** [comment]
I assume your goal is to run the game on an Android phone? I should probably just release an Android port at some point if people really want something like that.

**Comment by dezmen3** [comment]
> I assume your goal is to run the game on an Android phone? I should probably just release an Android port at some point if people really want something like that.

If you can, that would be more than awesome! In the era of android handhelds we need this for sure

**Comment by DTK21** [comment]
[](https://www.reddit.com/r/yuzuAndroid/comments/1726ksv/yuzu_android/)

It definitely works on Yuzu Android, as long as you set you're own NSP forwarder up, use a clean install of the game, and put all files needed inside the Yuzu Android Emu's SDMC\Switch Folder. The forwarder NSP should be where you usually put other switch games

**Comment by ivan89el** [comment]
Emulators are crutches. I would like a native port (I have a 4:3 640x480 device)

**Comment by DTK21** [comment]
> > It definitely works on Yuzu Android, as long as you set you're own NSP forwarder up, use a clean install of the game, and put all files needed inside the Yuzu Android Emu's SDMC\Switch Folder. The forwarder NSP should be where you usually put other switch games
> 
> That's how I did basically.
> 
> Unpacked Switch folder to yuzu app / files / sdmc / And placed The Simpsons_ Hit & Run (port) [018fcc923d8d0000].nsp where my all nsp located. Crash on start

Did you make the forwarder yourself, with the correct .nro name and folder pathing and name?
If so then what did you put down for your nro forwarder homebrew path?

**Comment by dezmen3** [comment]
> > > It definitely works on Yuzu Android, as long as you set you're own NSP forwarder up, use a clean install of the game, and put all files needed inside the Yuzu Android Emu's SDMC\Switch Folder. The forwarder NSP should be where you usually put other switch games
> > 
> > 
> > That's how I did basically.
> > Unpacked Switch folder to yuzu app / files / sdmc / And placed The Simpsons_ Hit & Run (port) [018fcc923d8d0000].nsp where my all nsp located. Crash on start
> 
> Did you make the forwarder yourself, with the correct .nro name and folder pathing and name? If so then what did you put down for your nro forwarder homebrew path?

Have you launched Hit&Run on yuzu android?

**Comment by DTK21** [comment]
Yes, the only time I got it to work was when I made the NSP forwarder myself using a tool
I used the one here [](https://nsp-forwarder.n8.io/)

**Comment by dezmen3** [comment]
> Yes, the only time I got it to work was when I made the NSP forwarder myself using a tool I used the one here [https://nsp-forwarder.n8.io/](url) I'd advice you to use the advanced mode just to make sure to have a random title ID, accurate version numbering, and the other options

That sounds over-complicated for end-user

**Comment by ZenoArrows** [comment]
I've pushed the source code now, if anyone is interested in helping out with an Android port they're welcome to contribute.

**Comment by dezmen3** [comment]
> I've pushed the source code now, if anyone is interested in helping out with an Android port they're welcome to contribute.

the GOAT! :)

**Comment by DTK21** [comment]
> > Yes, the only time I got it to work was when I made the NSP forwarder myself using a tool I used the one here [https://nsp-forwarder.n8.io/](url) I'd advice you to use the advanced mode just to make sure to have a random title ID, accurate version numbering, and the other options
> 
> That sounds over-complicated for end-user

I can assure you it's as simple as dragging the .nro file into the tool, setting the correct folder name the files are in and bam you have your own NSP forwarder instead of relying on someone else's that may have a different folder name.
Also me suggesting the use of Advanced mode in the tool wasn't mandatory in any way, just a recommendation just to be clear

**Comment by ZenoArrows** [comment]
@iscle Add me on discord if you need some help, I can probably offer some helpful knowledge about the codebase. My username is zenoselea.

**Comment by iscle** [comment]
> @iscle Add me on discord if you need some help, I can probably offer some helpful knowledge about the codebase. My username is zenoselea.

Added! My username is iscle :)
